openretina/cli/train.py

```diff
@@ -87,6 +102,50 @@ def train_model(cfg: DictConfig) -> None:
     dataloader_mapping = {f"DataLoader {i}": x[0] for i, x in enumerate(short_cyclers)}
     log.info(f"Dataloader mapping: {dataloader_mapping}")
     trainer.test(model, dataloaders=[c for _, c in short_cyclers], ckpt_path="best")
+    # Check if MLflow is one of the loggers and save model and datasets as artifacts
+    for logger in logger_array:
+        if "mlflow" in str(type(logger)).lower():
+            try:
+                # Get the existing run from the Lightning MLflow logger
+                run_id = logger.run_id
+
+                # Use the active run context
+                with mlflow.start_run(run_id=run_id, nested=True) as run:
+                    # Log the model as an artifact
+                    mlflow.pytorch.log_model(model, "model")
+                    log.info("Logged model to MLflow")
+
+                    # Log dataset information
+                    mlflow.log_dict(data_info, "data_info.json")
+                    log.info("Logged data information to MLflow")
+
+                    # Log the full configuration
+                    mlflow.log_dict(OmegaConf.to_container(cfg, resolve=True), "config.json")
+
+                    # Log system information
+                    sys_info = {
+                        "python_version": sys.version,
+                        "platform": platform.platform(),
+                        "lightning_version": lightning.__version__,
+                    }
+                    mlflow.log_dict(sys_info, "system_info.json")
+
+                    # Log model summary if available
+                    if hasattr(model, "summarize"):
+                        mlflow.log_text(str(model.summarize()), "model_summary.txt")
+            except Exception as e:
+                log.warning(f"Failed to log artifacts to MLflow: {str(e)}")
```

Comment:
```suggestion
                log.exception(f"Failed to log artifacts to MLflow: {str(e)}")
```
If you use log.exception it will log the whole stack trace. Not sure if you want this here, but it's useful to know

Reply:
I don't think we want the whole trace since I wouldn't consider MLFlow a priority in the log. But good to know for sure.